### PR TITLE
Add FarmActivity.kt

### PR DESCRIPTION
**Pull request by AliaksandrG**

### CODE REVIEW

File: app/src/main/java/com/example/myapplication/FarmActivity.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_farm)
+
+        buttonStartFarm.setOnClickListener {
+            isHarvest()
+        }
+    }
+
+    private fun isHarvest() = runBlocking<Unit> {
```
Comment (stasenkots) [inline app/src/main/java/com/example/myapplication/FarmActivity.kt:20]: don't use runBlocking

**Review by stasenkots — CHANGES_REQUESTED** [review]
You should use ViewModel in the task.And coroutines for emulating "receiving some data from server".also better to group repeated logic

File: app/src/main/java/com/example/myapplication/FarmActivity.kt (hunk 1 above)
Comment (AliaksandrG) [reply app/src/main/java/com/example/myapplication/FarmActivity.kt]: fixed

**Review by stasenkots — APPROVED** [review]

**Review by PolzaAnt — APPROVED** [review]

File: app/src/main/java/com/example/myapplication/FarmActivity.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import com.example.myapplication.countryModel.MyCountryModel
+import kotlinx.android.synthetic.main.activity_farm.*
+
+class FarmActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_farm)
+
+        val viewModel: MyCountryModel by lazy {
```
Comment (TMSlebedev) [inline app/src/main/java/com/example/myapplication/FarmActivity.kt:18]: nice

File: app/src/main/java/com/example/myapplication/FarmActivity.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+            viewModel.isHarvest()
+
+            viewModel.winner.observe(this, Observer { winner ->
+                Toast.makeText(this, "Победил $winner", Toast.LENGTH_LONG).show()
+            })
+        }
+
+        buttonResetFarmData.setOnClickListener {
+            CountryObject.Gomel.cabbage.value = 0
```
Comment (TMSlebedev) [inline app/src/main/java/com/example/myapplication/FarmActivity.kt:55]: use postValue

File: app/src/main/java/com/example/myapplication/countryModel/MyCountryModel.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+            while (true) {
+                for (country in countryList) {
+                    country.apply {
+                        potato.postValue(potato.value?.plus(Random.nextInt(10)))
+                        beet.postValue(beet.value?.plus(Random.nextInt(10)))
+                        cabbage.postValue(cabbage.value?.plus(Random.nextInt(10)))
+                    }
+
+                    if (country.potato.value!! == 100
```
Comment (TMSlebedev) [inline app/src/main/java/com/example/myapplication/countryModel/MyCountryModel.kt:27]: don't use asser calls '!!' use safe requests with '?'

File: app/src/main/res/layout/activity_farm.xml [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+            android:layout_height="match_parent"
+            android:id="@+id/gomel_cabbage"
+            android:text="0"
+            android:gravity="center"
+            android:textSize="40sp"/>
+    </RelativeLayout>
+
+
+
+
```
Comment (TMSlebedev) [inline app/src/main/res/layout/activity_farm.xml:140]: delete empty

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -31,9 +31,10 @@
         app:layout_constraintVertical_bias="0.0" />
 
     <Button
+        android:id="@+id/buttonHWTwenty"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Button3"
+        android:text="homework 20"
```
Comment (TMSlebedev) [inline app/src/main/res/layout/activity_main.xml:37]: add to Strings

**Review by TMSlebedev — APPROVED** [review]
Fix selected and merge

**Pull request merged**